### ISSUE TITLE
MSVC fails to build and detect compilers for vs preview after #91069

### ISSUE THREAD

**Issue by HKunogi** [godotengine/godot#97619]
### Tested versions

After [this commit](https://github.com/godotengine/godot/pull/91069) it fails to compile while using vs studio preview, the change on methods.py should take into account for preview versions by using the '-prerelease' flag on the command, and also take into account the return of the 'Preview' keyword on the result to properly parse the returned value.

### System information

Windows 11

### Issue description

Cannot build due to failure on detecting compiler when using vs studio preview, it used to work prior to the specified commit.

### Steps to reproduce

Simply install vs studio preview (without any other versions installed) and try to build it using the scons script

### Minimal reproduction project (MRP)

Build issue, so theres no MRP

**Comment by akien-mga** [comment]
Thanks for the report. It's a duplicate of #97614 so I'll consolidate it there. A fix will be merged shortly.

**Comment by akien-mga** [comment]
I misunderstood, this isn't a duplicate but specific to VS Preview.
